analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -147,4 +147,23 @@ public override void ExecutionCompleted()
             ReportIssue(operation, operation.Syntax.ToString());
         }
     }
+
+    private static ProgramState SetConstraintForObjectCreation(ProgramState state, IObjectCreationOperationWrapper objectCreation) =>
+        FirstArgumentAsCollection(objectCreation) switch
+        {
+            { } firstArgument when ConstraintFromArgument(state, firstArgument) is { } constraint => state.SetOperationConstraint(objectCreation, constraint),
+            { } => state,
+            _ => state.SetOperationConstraint(objectCreation, CollectionConstraint.Empty)
+        };
+
+    private static IArgumentOperationWrapper? FirstArgumentAsCollection(IObjectCreationOperationWrapper objectCreation) =>
+        objectCreation.Arguments.Select(x => x.AsArgument()).SingleOrDefault(x => x?.Parameter.Ordinal == 0) is { } firstArgument
+        && firstArgument.Parameter.Type.DerivesOrImplements(KnownType.System_Collections_IEnumerable)
+        ? firstArgument
+        : null;
+
+    private static CollectionConstraint ConstraintFromArgument(ProgramState state, IArgumentOperationWrapper argument) =>
+        argument.WrappedOperation.TrackedSymbol() is { } symbol
+        ? state[symbol]?.Constraint<CollectionConstraint>()
+        : null;
```

Comment:
This should not look at symbol, but the operation itself. And after that, this should not be embedded.
```suggestion
        state[argument]?.Constraint<CollectionConstraint>()
```

Reply:
This does not work. `argument` doesn't seem to have a constraint even in very simple cases.

Reply:
The `return` of `Argument` processor should be likely like this:
```
        return state[argument.Value] is { } value
            ? state.SetOperationValue(argument, value)
            : state;
```

Reply:
This worked, nice. It does not fix the `new List<int>(condition ? empty : empty)` FN, though.